api/src/pcapi/core/users/api.py

```diff
@@ -1019,6 +1019,19 @@ def _filter_user_accounts(
     if search_term.isnumeric():
         term_filters.append(models.User.id == int(search_term))
 
+    # birthdate
+    try:
+        date_term = datetime.datetime.strptime(search_term, "%d/%m/%Y")
+    except ValueError:
+        pass  # date has invalid format
+    else:
+        term_filters.append(
+            sa.or_(
+                models.User.dateOfBirth == date_term,
+                models.User.validatedBirthDate == date_term,
+            )
+        )
+
```

Comment:
Est-ce que filtrer sur birth_date irait ?
```
@property
    def birth_date(self) -> date | None:
        """
        Returns the birth date validated by an Identity Provider if it exists,
        otherwise the birth date declared by the user at signup.
        """
        if self.validatedBirthDate:
            return self.validatedBirthDate
        if self.dateOfBirth:
            return self.dateOfBirth.date()
        return None
```

Reply:
`User.birth_date` est une property (Python pur, donc), pas une _hybrid property_ utilisable pour construire une requête SQL. Tu peux essayer des variantes de ta requête en SQL sur staging.

(**Correction de mon message initial :** on a bien un index sur `validatedBirthDate`. Donc une recherche sur cette colonne-là a l'air d'être raisonnablement rapide.)

Reply:
Après étude, `validatedBirthDate` est suffisant au besoin